erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -126,11 +139,41 @@ def validate_item(self):
 				frappe.throw(_("Row #{0}: Please specify Serial No for Item {1}").format(item.idx, item.item_code),
 					frappe.MandatoryError)
 
+	def validate_qty(self):
+		manufacture_purpose = ["Manufacture", "Material Consumption for Manufacture"]
+
+		if self.purpose in manufacture_purpose and self.work_order:
+			if not frappe.get_value('Work Order', self.work_order, 'skip_transfer'):
+				item_code = []
+				for item in self.items:
+					req_items = frappe.get_all('Work Order Item',
+									filters={'parent': self.work_order, 'item_code': item.item_code}, fields=["item_code"])
+
+					transferred_materials = frappe.db.sql("""
+								select
+									sum(qty) as qty
+								from `tabStock Entry` se,`tabStock Entry Detail` sed
+								where
+									se.name = sed.parent and se.docstatus=1 and
+									(se.purpose='Material Transfer for Manufacture' or se.purpose='Manufacture')
+									and sed.item_code=%s and se.work_order= %s and ifnull(sed.t_warehouse, '') != ''
+							""", (item.item_code, self.work_order), as_dict=1)
+
+					stock_qty = flt(item.qty)
+					trans_qty = flt(transferred_materials[0].qty)
+					if req_items:
+						if stock_qty > trans_qty:
+							item_code.append(item.item_code)
+				if item_code:
+					frappe.throw(_("Consumption quantity cannot be more than transferred quantity for item/s {0}.").format(', '.join(item_code)), OverConsumptionError)
```

Comment:
Should we do this validation? Because many cases, some unused raw materials (from earlier Work Orders) are there in WIP warehouse, which gets used. And it can lead to more consumption than transferred qty.